annotated.go

```diff
@@ -52,6 +52,8 @@ import (
 //   })
 //
 // Annotated cannot be used with constructors which produce fx.Out objects.
+//
+// When used with fx.Supply, the target is a value rather than a constructor function.
```

Comment:
Would it make sense to separate the two? I think we're still in the `is the function a value as well` dilemma that is being solved by separating `fx.Supply` from just extending the `fx.Provide`, right?

I think it could make sense to add another member like `Value interface{}`

Reply:
The function-value ambiguity isn't an issue with `fx.Annotate`, as the interpretation of `Target` mirrors the option function it's used with: `fx.Provide` expects `Target` to be a function (proven by https://github.com/uber-go/fx/blob/master/app_test.go#L199), and `fx.Supply` will always wrap the target.

Reply:
FWIW we named the field `Target` instead of constructor specifically so that it could have values in the future. The original implementation named it something like `Func` or `Constructor`.